### ISSUE TITLE
Replacement suggested by manual_assert discards attributes

### ISSUE THREAD

**Issue by dtolnay** [rust-lang/rust-clippy#7935]
```rust
fn main() {
    let x = 0;
    if x == 0 {
        #[cfg(not(feature = "less_crashy"))]
        panic!("huh");
    }
}
```

```console
$ cargo clippy -- -Wclippy::pedantic
warning: only a `panic!` in `if`-then statement
 --> src/main.rs:3:5
  |
3 | /     if x == 0 {
4 | |         #[cfg(not(feature = "less_crashy"))]
5 | |         panic!("huh");
6 | |     }
  | |_____^ help: try: `assert!(!(x == 0), "huh");`
  |
  = note: `-W clippy::manual-assert` implied by `-W clippy::pedantic`
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#manual_assert
```

Would it be possible to suggest the following?

```diff
+ #[cfg(not(feature = "less_crashy"))]
+ assert!(..., "huh");
```

In fact this may not be equivalent if the branch condition has side effects, so maybe we should even consider this a false positive and not lint at all.

### Meta

**Rust version (`rustc -Vv`):**

```
rustc 1.58.0-nightly (4961b107f 2021-11-04)
binary: rustc
commit-hash: 4961b107f204e15b26961eab0685df6be3ab03c6
commit-date: 2021-11-04
host: x86_64-unknown-linux-gnu
release: 1.58.0-nightly
LLVM version: 13.0.0
```
